### PR TITLE
fix(files): Only add copy suffix before file extension for files (not folders)

### LINKED ISSUE #43843: [Bug]: String `(copy)` misplaced with some filenames
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When you copy a File to the same folder as the original file is, the string "(copy)" is added to the file name.
If the filename consists a "." the "(copy)" is missplaced.

file.name.txt -> file (copy).name.txt

### Steps to reproduce

1. copy a File where filename consists a "." e.g. "file.name.txt"
2. copy to the same folder as origin
3. The copied file is renamed file.name.txt -> file (copy).name.txt

### Expected behavior

file.name.txt -> file.name (copy).txt

### Installation method

Other Community project

### Nextcloud Server version

28

### Operating system

Debian/Ubuntu

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by susnux**
* Resolves: https://github.com/nextcloud/server/issues/43843

## Summary

For files keep the behavior (`file.name` -> `file (copy).name`) but for folders ignore the extension (`folder.name` -> `folder.name (copy)`).

https://github.com/nextcloud/server/assets/1855448/f159df0e-eedc-45a2-8060-e3f64744b9a6

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files/src/utils/fileUtils.ts [hunk 1]
```diff
@@ -28,15 +28,24 @@ import { translate as t, translatePlural as n } from '@nextcloud/l10n'
...
  * @param name The initial name to use
  * @param otherNames Other names that are already used
- * @param suffix A function that takes an index an returns a suffix to add, defaults to '(index)'
+ * @param options Optional parameters for tuning the behavior
+ * @param options.suffix A function that takes an index and returns a suffix to add to the file name, defaults to '(index)'
+ * @param options.ignoreFileExtension Set to true to ignore the file extension when adding the suffix (when getting a unique directory name)
  * @return Either the initial name, if unique, or the name with the suffix so that the name is unique
  */
-export const getUniqueName = (name: string, otherNames: string[], suffix = (n: number) => `(${n})`): string => {
+export const getUniqueName = (name: string, otherNames: string[], options: { suffix?: (i: number) => string, ignoreFileExtension?: boolean } = {}): string => {
```
Comment (Pytal) [inline apps/files/src/utils/fileUtils.ts:36]:
```suggestion
export const getUniqueName = (
  name: string, 
  otherNames: string[], 
  options: { 
    suffix?: (i: number) => string, 
    ignoreFileExtension?: boolean,
  } = {},
): string => {
```
More readable?

**Review by Pytal — APPROVED** [review]

File: apps/files/src/utils/fileUtils.ts (hunk 1 above)
Comment (susnux) [reply apps/files/src/utils/fileUtils.ts:36]: looks good!

File: apps/files/src/utils/fileUtils.ts [hunk 2]
```diff
@@ -28,15 +28,31 @@ import { translate as t, translatePlural as n } from '@nextcloud/l10n'
...
  * @param name The initial name to use
  * @param otherNames Other names that are already used
- * @param suffix A function that takes an index an returns a suffix to add, defaults to '(index)'
+ * @param options Optional parameters for tuning the behavior
+ * @param options.suffix A function that takes an index and returns a suffix to add to the file name, defaults to '(index)'
+ * @param options.ignoreFileExtension Set to true to ignore the file extension when adding the suffix (when getting a unique directory name)
  * @return Either the initial name, if unique, or the name with the suffix so that the name is unique
  */
-export const getUniqueName = (name: string, otherNames: string[], suffix = (n: number) => `(${n})`): string => {
+export const getUniqueName = (
```
Comment (skjnldsv) [inline apps/files/src/utils/fileUtils.ts:36]: todo: move to `@nextcloud/vue` :see_no_evil:

Comment (susnux) [reply apps/files/src/utils/fileUtils.ts:36]: `@nextcloud/files` ?

Comment (skjnldsv) [reply apps/files/src/utils/fileUtils.ts:36]: yes, brain died while writing my message :see_no_evil:

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**